src/services/preProcess.ts

```diff
@@ -78,7 +78,7 @@ namespace ts {
          */
         function tryConsumeImport(): boolean {
             let token = scanner.getToken();
-            if (token === SyntaxKind.ImportKeyword) {
+            if (token === SyntaxKind.ImportKeyword && !scanner.hasPrecedingDot()) {

```

Comment:
do not think you need to change the scanner API to do this. this whole function is just a mini parser. it advances the scanner each time it needs to consume a token, so you can do it here, no need to change the scanner implementation.

Reply:
Okay, sorry for my wording, I used `parser`, because you called it a `mini parser`. So flag in `nextToken` SGTY?

Reply:
yeah. it is already keeping state, one more flag should not harm anyone.

Reply:
Ok, will give it a spin 👍

Reply:
changes in ee429ef, but for some reason it doesn't work as expected. I am not even sure if the code is run, as I can't figure out from the docs on how to debug the tests or use the watch mode. Plus, `process.stdout.write` and `console.log` are not available either, so I am unsure on how to debug this - any pointers @mhegazy?